zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -2026,6 +2031,87 @@ public void getData(final String path, Watcher watcher, DataCallback cb, Object
         cnxn.queuePacket(h, new ReplyHeader(), request, response, cb, clientPath, serverPath, ctx, wcb);
     }
 
+    /**
+     * TODO
+     * @param path
+     * @param watcher
+     * @param readMode
+     * @return
+     * @throws KeeperException
+     * @throws InterruptedException
+     * @since
+     */
+    public CompletableFuture<ZNode> getData(ReadConsistencyMode readMode, final String path, Watcher watcher) throws KeeperException, InterruptedException {
```

Comment:
CompletableFuture is tricky, because there is no control over the thread that execute any callback/listener

it is appealing but I won't use it here, all the API is based on callbacks that are guaranteed to be executed in the ZK thread